src/peergos/server/net/DHTHandler.java

```diff
@@ -1,34 +1,39 @@
 package peergos.server.net;
 
+import peergos.server.storage.*;
 import peergos.shared.cbor.*;
 import peergos.shared.crypto.asymmetric.*;
 import peergos.shared.crypto.hash.*;
 import peergos.shared.io.ipfs.api.*;
 import peergos.shared.io.ipfs.multihash.*;
 import peergos.shared.io.ipfs.cid.*;
+import peergos.shared.mutable.*;
 import peergos.shared.storage.ContentAddressedStorage;
 import com.sun.net.httpserver.*;
 import peergos.shared.util.*;
 
 import java.io.*;
 import java.util.*;
+import java.util.concurrent.*;
 import java.util.function.*;
 import java.util.stream.*;
 
 public class DHTHandler implements HttpHandler
 {
     private static final boolean LOGGING = true;
     private final ContentAddressedStorage dht;
+    private final KeyFilter keyFilter;
     private final String apiPrefix;
 
-    public DHTHandler(ContentAddressedStorage dht, String apiPrefix) throws IOException
+    public DHTHandler(ContentAddressedStorage dht, KeyFilter keyFilter, String apiPrefix) throws IOException
     {
         this.dht = dht;
+        this.keyFilter = keyFilter;
         this.apiPrefix = apiPrefix;
     }
 
-    public DHTHandler(ContentAddressedStorage dht) throws IOException {
-        this(dht, "/api/v0/");
+    public DHTHandler(ContentAddressedStorage dht, KeyFilter keyFilter) throws IOException {
+        this(dht, keyFilter, "/api/v0/");
     }
 
     private Map<String, List<String>> parseQuery(String query) {
```

Comment:
Nit:  this function is fairly long, having sub functions for each case would improve readability.

Reply:
Agreed.